canonical-web-and-design/ubuntu.com
# USN API endpoint script

## Done

- Update the post USN endpoint to extract the USN ID from the payload itself  so we don't have to ley the payload on the USN ID.
- Add a script to hit the endpoint. It currently accepts files that look like https://people.canonical.com/~ubuntu-security/usn/database.json.

## QA

- `dotrun clean`
- `dotrun exec alembic upgrade head`
- `dotrun exec python3 webapp/security/fixtures/releases.py`
- `curl https://people.canonical.com/~ubuntu-security/usn/database.json > webapp/security/fixtures/usns.json`
- `dotrun`
- in a separate tab `dotrun exec python3 webapp/security/fixtures/usns.py webapp/security/fixtures/usns.json`
- feel free to pause script execution as there are a lot of USNs on that database.
- go to `/security/notices` and verify there are listed USNs

## Issue / Card

Fixes https://github.com/canonical-web-and-design/web-squad/issues/2637
